graylog2-server/src/main/java/org/graylog2/shared/journal/KafkaJournal.java

```diff
@@ -168,6 +170,9 @@ public KafkaJournal(@Named("message_journal_dir") File journalDirectory,
         this.scheduler = scheduler;
         this.throttleThresholdPercentage = intRange(throttleThresholdPercentage, 0, 100);
         this.serverStatus = serverStatus;
+        this.maxSegmentSize = segmentSize.toBytes();
+        // Max message size should not be bigger than max segment size.
+        this.maxMessageSize = Math.min(Ints.saturatedCast(maxSegmentSize), Integer.MAX_VALUE);
```

Comment:
`Ints#saturatedCast()` will always be less than or equal to `Integer.MAX_VALUE`.